core/src/lib.rs

```diff
@@ -63,15 +66,46 @@ impl Raiko {
     pub async fn generate_input<BDP: BlockDataProvider>(
         &self,
         provider: BDP,
+    ) -> RaikoResult<GuestInput> {
+        if self.v2_preflight {
+            self.generate_input_v2(provider).await
+        } else {
+            self.generate_input_v1(provider).await
+        }
```

Comment:
What is the difference between the preflight_v1() and preflight_v2()? What is the rationale behind dividing them into two separate functions?

Here is my consideration. I have tried to diff preflight_v1() and preflight_v2() and found that they are nearly the same. So I think we may not to divide them.

Also, generate_input_v1() and generate_input_v2() are in the same issue.

Reply:
Yes, so far they are nearly the same, I separated for easier code impl for now, after node running, there will be some updates in execute_tx part, will see if the diff is big enough or not later on.